agent/cache/cache.go

```diff
@@ -396,20 +414,20 @@ func makeEntryKey(t, dc, token, key string) string {
 // If allowNew is true then the fetch should create the cache entry
 // if it doesn't exist. If this is false, then fetch will do nothing
 // if the entry doesn't exist. This latter case is to support refreshing.
-func (c *Cache) fetch(tEntry typeEntry, key string, r Request, allowNew bool, attempt uint, minIndex uint64, ignoreExisting bool, ignoreRevalidation bool) (<-chan struct{}, error) {
-	info := r.CacheInfo()
+func (c *Cache) fetch(key string, r request, allowNew bool, attempt uint, ignoreExisting bool) <-chan struct{} {
```

Comment:
There is more that can be done here, but I've left it for a follow up. There are still 2 boolean args, and the callers always use either `true, false`, and `false, true`. We could represent it with a single boolean, or even better we could extract the functionality into two functions and pass in those functions to replace the bools.

I considered doing that, but I think a better option is going to be to split `fetch` into two functions. I think we may be able to do that along with the introduction of `x/sync/singleflight`.

Reply:
Something [like this](https://github.com/hashicorp/consul/compare/dnephin/agent-cache-request-info...dnephin/agent-cache-fetch). It doesn't yet add singleflight, but shows that a significant number of branches in fetch are not actually shared between the two callers.